internal/service/ec2/network_insights_analysis.go

```diff
@@ -0,0 +1,199 @@
+package ec2
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/aws-sdk-go-base/v2/awsv1shim/v2/tfawserr"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/customdiff"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	tftags "github.com/hashicorp/terraform-provider-aws/internal/tags"
+	"github.com/hashicorp/terraform-provider-aws/internal/tfresource"
+	"github.com/hashicorp/terraform-provider-aws/internal/verify"
+)
+
+func ResourceNetworkInsightsAnalysis() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceNetworkInsightsAnalysisCreate,
+		ReadContext:   resourceNetworkInsightsAnalysisRead,
+		UpdateContext: resourceNetworkInsightsAnalysisUpdate,
+		DeleteContext: resourceNetworkInsightsAnalysisDelete,
```

Comment:
In my last PR these were changed to `WithoutTimeout` variants by a maintainer, but I am not sure why (I would like to learn though!). I _do_ use timeouts in this PR so figured not to do the same here.

Reply:
The `Context` (not `WithoutTimeout`) methods set a context timeout for each resource CRUD method from either the per-operation custom timeout block or to a 20 minute default value. This timeout applies to all the AWS API calls made in the method and all the `WaitForState` waiters, not each call/waiter in turn. Because this is a potentially breaking change in behavior the `WithoutTimeout` methods were added. These methods set no context timeout and allow the WaitForState` waiter` timeout to apply to each waiter.
See https://github.com/hashicorp/terraform-provider-aws/issues/15090 for a discussion.